docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -92,40 +94,41 @@ spec:
     ports:
       - containerPort: 443
 ```
-When a pod is attached to a certain network, it needs to set its "network" field:
+When a pod is attached to a certain network, it needs to set its "network" in labels:
 
 ```yaml
 apiVersion: v1
 kind: Pod
 metadata:
   name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
 spec:
   containers:
   - name: nginx
     image: nginx
     ports:
       - containerPort: 443
-  network: vpc-1
 ```
 
 
-When a pod is attached to a certain network and it wants to specify subnet or IP, it needs to be specified in pod fields:
+When a pod is attached to a certain network and it wants to specify subnet or IP, it can be expressed in annotations:
 
 ```yaml
 apiVersion: v1
 kind: Pod
 metadata:
   name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
+  annotations:
+    arktos.futurewei.com/nics: [{"name": "eth0", "ip": "192.168.0.12"}]
```

Comment:
Since nics only specify the ethernet information here, how is subnet name gets passed to network provider? 
The pod is associated with Subnets, not VPC. Later in the doc, you also mentioned a CIDR range, is that the subnet's CIDR range?

Reply:
this exemplary line only reflects subset that nics could have.  please refer to [previous network extension design](https://github.com/futurewei-cloud/arktos/blob/master/docs/design-proposals/network/NICAndVPCSupportInArktos.md), pod spec change section to see what others may be supported. One valid example is {"name":"eth0", "subnet":"my-subnet-name"}. What in this nics annotation is intended for user to specify some finer constraint beyond the vpc defined in network object; nics info is optional.
From Arktos perspective, pod associated to an IP address. For vpc typed network, of course it must be on a VPC network.

As for the cidr (in vip-hint annotation, and network.spec), it is for service objects, not pod objects. This is extended from cluster-ip-range of k8s parameter, which is to define the range of service virtual IP addresses.

Please be noted nics annotation does not mandate subnet cannot have value in cidr form; if it makes sense, it can has cidr value instead of subnet name. It depends on what external network provider understands.

Reply:
I now understand the reason you have the nice. In this case, for PODS, the NICS is the one we really care about. So doesn't it make sense then that all the network specific fields goes to the NIC. For example, a POD can have multiple NICs each belong to a different VPC/Subnet? Also the IP address allocation on Mizar side, will be NIC specific not POD specific. 

Also, how the user will be able to add/remove NICs from the POD. How are you going to (re)-invoke the CNI workflow then?

Reply:
Interesting concept here, with possibly multiple network interfaces linked to a pod and each network interface can be in different subnet. If IP address allocation is NIC specific, we can add a new Nic type object in Mizar, and after Nic object is provisioned, it then will be associated with a pod.

Reply:
it seems too much for us to implement multiple nics right now. Let's focus on single nic right now.
I'll update design to change nics to nic only to avoid confusion.
If we decide to support multiple nics in the future, we can pick up annotation nics by that time.

Reply:
We need also make sure that there is at least one subnet within any of the vpc listed under vpcID field! Otherwise, if user does not specify a subnet, we cannot create the pod as the pod needs to be associated and only associated with a subnet.

Reply:
subnets's creation should be totally managed by network provider. user does not have to specify the subnet - it is optional as hint if present, not mandatory. Hope that makes sense.

Reply:
Yeah, that's fine. My point is if vpc listed does not have a subnet, Mizar will throw an error saying no subnet in vpc. Does that make sense?

Reply:
based on our discussion a few days ago, mizar network should exists __before__ arktos network object is created. I think mizar network should include valid subnet in place.

Reply:
Customer needs to specify the subnet CIDR range, thus customer **has to** create a subnet manually. Mizar can't decide for customer subnet's CIDR.

Reply:
it is Okay that user has to ensure mizar vpc + subnet created before hand.
creating mizar vpc & subnets inside vpc is out of the design proposal's scope.